src/history/hs.c

```diff
@@ -823,6 +861,28 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
 
         if (modifies.size > 0)
             WT_STAT_CONN_INCR(session, cache_hs_write_squash);
+
+        /*
+         * We need to clear the history store if we haven't inserted anything into the history store
```

Comment:
Am confused with "We need to clear the history store if we haven't inserted anything into the history store"

Reply:
The change is to make sure we only need to delete things from the history store once even we have multiple mixed mode operations. Generally, we delete hs content when we insert the first mixed mode update to the history store. However, if the mixed mode operation is onpage or it is a tombstone, we may not insert anything into the history store but we still need to remove existing content in the history store. The code is cover these cases.